From f98b6a14b88b44b6adb54ae6d50d86b96c2a6736 Mon Sep 17 00:00:00 2001
From: Jason <oblitorum@gmail.com>
Date: Sat, 30 Mar 2024 03:53:50 +0800
Subject: [PATCH] Allow building with flto flag

---
 cmake/libutils.cmake | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/cmake/libutils.cmake b/cmake/libutils.cmake
index 0f8786fc264..eeb0a00a28c 100644
--- a/cmake/libutils.cmake
+++ b/cmake/libutils.cmake
@@ -87,11 +87,6 @@ MACRO(CREATE_EXPORTS_FILE VAR TARGET API_FUNCTIONS)
     ENDFOREACH()
     SET(CONTENT "${CONTENT} (void *)0\n}\;")
     CONFIGURE_FILE_CONTENT(${CONTENT} ${EXPORTS})
-    # Avoid "function redeclared as variable" error
-    # when using gcc/clang option -flto(link time optimization)
-    IF(" ${CMAKE_C_FLAGS} ${CMAKE_CXX_FLAGS} " MATCHES " -flto")
-      SET_SOURCE_FILES_PROPERTIES(${EXPORTS} PROPERTIES COMPILE_FLAGS "-fno-lto")
-    ENDIF()
     SET(${VAR} ${EXPORTS})
   ENDIF()
 ENDMACRO()
-- 
2.25.1

